lib/profile_test.go

```diff
@@ -95,20 +93,9 @@ func TestProfileRequestsSave(t *testing.T) {
 	}
 }
 
-func TestSaveProfile(t *testing.T) {
-	ctx, done := context.WithCancel(context.Background())
-	defer done()
-
-	cfg := testcfg.DefaultConfigForTesting()
-
-	// Mock data for the global Config's Profile, used to create new profile.
-	// TODO: Remove the randomly built Profile that config.DefaultProfile creates.
-	mockID := "QmWu3MKx2B1xxphkSNWxw8TYt41HnXD8R85Kt2UzKzpGH9"
-	mockTime := time.Unix(1234567890, 0)
-	cfg.Profile.ID = mockID
-	cfg.Profile.Created = mockTime
-	cfg.Profile.Updated = mockTime
-	cfg.Profile.Peername = "test_mock_peer_name"
+func TestSetProfile(t *testing.T) {
+	tr := newTestRunner(t)
+	defer tr.Delete()
```

Comment:
using the old `NewInstanceFromConfigAndNode` was actually messing with our expectations, since `NewInstanceFromConfigAndNode` does it's own strange config magic. using `testRunner` instead.